src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs

```diff
@@ -102,42 +101,21 @@ private Manifest BuildCore(DocumentBuildParameters parameters)
                     parameters.VersionName,
                     parameters.ApplyTemplateSettings,
                     parameters.RootTocPath);
-                IHostServiceCreator hostServiceCreator = null;
-                PhaseProcessor phaseProcessor = null;
-                if (ShouldTraceIncrementalInfo)
-                {
-                    context.IncrementalBuildContext = IncrementalBuildContext.Create(parameters, CurrentBuildInfo, LastBuildInfo, IntermediateFolder);
-                    hostServiceCreator = new HostServiceCreatorWithIncremental(context);
-                    phaseProcessor = new PhaseProcessor
-                    {
-                        Handlers =
-                        {
-                            new PrebuildBuildPhaseHandlerWithIncremental(context),
-                            new PostbuildPhaseHandlerWithIncremental(context),
-                        }
-                    };
-                }
-                else
-                {
-                    hostServiceCreator = new HostServiceCreator(context);
-                    phaseProcessor = new PhaseProcessor
-                    {
-                        Handlers =
-                        {
-                            new PrebuildBuildPhaseHandler(context),
-                            new PostbuildPhaseHandler(context),
-                        }
-                    };
-                }
 
                 Logger.LogVerbose("Start building document...");
 
                 // Start building document...
                 List<HostService> hostServices = null;
+                IHostServiceCreator hostServiceCreator = null;
+                PhaseProcessor phaseProcessor = null;
                 try
                 {
                     using (var templateProcessor = parameters.TemplateManager?.GetTemplateProcessor(context, parameters.MaxParallelism) ?? TemplateProcessor.DefaultProcessor)
                     {
+                        using (new LoggerPhaseScope("Prepare", true))
+                        {
+                            Prepare(parameters, context, templateProcessor, out hostServiceCreator, out phaseProcessor);
```

Comment:
nice, see that

---
In reply to: [89715562](https://github.com/dotnet/docfx/pull/929#discussion_r89715562) [](ancestors = 89715562,89707493)